book.cmd/src/main/java/com/happystays/book/cmd/domain/BookAggregate.java

```diff
@@ -0,0 +1,30 @@
+package com.happystays.book.cmd.domain;
+
+import com.happystays.book.cmd.api.commands.BookCommand;
+import com.happystays.book.common.dto.*;
+import com.happystays.book.common.events.BookingSuccessEvent;
+import com.happystays.cqrs.core.domain.AggregateRoot;
+import com.happystays.cqrs.core.dto.BookResponse;
+import lombok.NoArgsConstructor;
+
+import java.util.Date;
+import java.time.Instant;
+import java.util.List;
+
+@NoArgsConstructor
+public class BookAggregate extends AggregateRoot {
+
+    public BookAggregate (BookResponse bookResponse , BookCommand command) {
+        raiseEvent(BookingSuccessEvent.builder()
+                .users(List.of())
+                .pnrList(List.of(new Pnr(null, Date.from(Instant.now()), bookResponse.getPnrInfo().getBookingState(), new Trip(bookResponse.getRoomRate().getTotalAmount(), bookResponse.getRoomRate().getCurrencyCode(), bookResponse.getStartDate(), bookResponse.getEndDate()), List.of(new HotelInfo(bookResponse.getRoomRate().getTotalAmount(), bookResponse.getRoomRate().getCurrencyCode(), bookResponse.getHotelCode(),
+                        bookResponse.getHotelName(), bookResponse.getHotelAddress(), command.getHotelPhone(), command.getCountryCode(), command.getCountryCode(), true,
+                        List.of(new HotelSegment(bookResponse.getRoomRate().getTotalAmount(), bookResponse.getRoomRate().getCurrencyCode(), bookResponse.getPnrInfo().getConfirmationNumber(), command.getGuestCount(), bookResponse.getStartDate(), bookResponse.getEndDate(), new HotelRoomInfo(0.0, "", command.isBreakfastIncluded()), new HotelCancellationInfo(command.getCancellationInfo().getCancellationDate(), true, command.getCancellationInfo().getCancellationPolicy(), command.getCancellationInfo().getCancellationDate()))),
+                        new PaymentMethod(command.getPaymentInfo().getPaymentType(), new CreditCard(command.getPaymentInfo().getCardNumber(), command.getPaymentInfo().getCardHolderName(), "")))), List.of())))
```

Comment:
userId you passed as empty string. We need to take that from auth token and set here. So can you put a comment with same.